src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -34,13 +34,9 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
                 {
                     clientManager = options.ClientManager;
                 }
-                else if (options.ConnectionStrings != null)
+                else if (options.ConnectionStrings != null || (options.Endpoints != null && options.Credential != null))
                 {
-                    clientManager = new ConfigurationClientManager(options.ConnectionStrings, options.ClientOptions);
-                }
-                else if (options.Endpoints != null && options.Credential != null)
-                {
-                    clientManager = new ConfigurationClientManager(options.Endpoints, options.Credential, options.ClientOptions);
+                    clientManager = new ConfigurationClientManager(options);
```

Comment:
`AzureAppConfigurationOptions` has a whole lot of things that `ConfigurationClientManager` doesn't need. There was a value not to just pass it down to ConfigurationClientManager.

I'd recommend having two constructors for ConfigurationClientManager. One for connection strings, and one for endpoints. I expect the code will end up cleaner as a result. Right now there must be a bit of a branched constructor that looks like two constructors put together.

Reply:
Sounds not bad